server/src/common/actions/organismes/organismes.duplicates.actions.ts

```diff
@@ -13,6 +11,34 @@ import { getEffectifsDuplicatesFromOrganismes } from "@/jobs/fiabilisation/uai-s
 export const getDuplicatesOrganismes = async () => {
   const duplicatesGroup = await organismesDb()
     .aggregate([
+      // Lookup sur les organisations
+      {
+        $lookup: {
+          from: "organisations",
+          as: "organisation",
+          let: {
+            uai: { $ifNull: ["$uai", null] }, // on force par défaut à null plutôt que undefined pour correspondre avec l'organisation
+            siret: "$siret",
+          },
+          pipeline: [{ $match: { $expr: { $and: [{ $eq: ["$siret", "$$siret"] }, { $eq: ["$uai", "$$uai"] }] } } }],
+        },
+      },
+      { $unwind: { path: "$organisation", preserveNullAndEmptyArrays: true } }, // preserveNullAndEmptyArrays pour garder les cas ou la jointure est vide
+
+      // Lookup sur les users depuis les organisations
+      {
+        $lookup: {
+          from: "usersMigration",
+          as: "users",
+          let: { organisation_id: "$organisation._id" },
+          pipeline: [
+            { $match: { $expr: { $eq: ["$organisation_id", "$$organisation_id"] } } },
+            { $project: { _id: 1 } },
+          ],
+        },
+      },
+
```

Comment:
On peut optimiser largement la requête en déplaçant le lookup derrière le $group. Chez moi on passe de 5.5s à 130ms.
Plutôt que de faire une jointure pour les 10k organismes, puis de grouper et filtrer, on récupère uniquement les 73 duplicats et puis on fait la jointure.

Faut juste s'assurer que le nombre d'utilisateurs est bien remonté (j'ai pas vérifié ça), mais vu qu'il y a des tests, c'est top. :+1:

Reply:
Je veux bien qu'on en discute car j'ai l'impression que c'est pas possible dans ce sens vu qu'on a besoin des users dans la liste "duplicates" 🤔

Reply:
Ah c'est vrai. Sorry.
Je serais quand même curieux de tester un lookup après coup, en itérant sur les duplicates. (puis à nouveau un $group pour réassembler.
Car je sens que les 5s chez moi vont donner au moins 10s en prod... :upside_down_face: 
On en reparle tout à l'heure !

Reply:
Oui ça c'est possible mais en terme de lisibilité de la requête ca pique lol.
J'ai testé avec la volumétrie de la prod en local c'est un peu long mais d'après moi acceptable pour l'instant.